main.go

```diff
@@ -0,0 +1,131 @@
+	var newLogger micrologger.Logger
+	{
+		loggerConfig := micrologger.DefaultConfig()
+		loggerConfig.IOWriter = os.Stdout
+		newLogger, err = micrologger.New(loggerConfig)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
+	// We define a server factory to create the custom server once all command
+	// line flags are parsed and all microservice configuration is storted out.
+	newServerFactory := func(v *viper.Viper) microserver.Server {
+		// Create a new custom service which implements business logic.
+		var newService *service.Service
+		{
+			serviceConfig := service.DefaultConfig()
+
+			serviceConfig.Flag = f
+			serviceConfig.Logger = newLogger
+			serviceConfig.Viper = v
+
+			serviceConfig.Description = description
+			serviceConfig.GitCommit = gitCommit
+			serviceConfig.Name = name
+			serviceConfig.Source = source
+
+			newService, err = service.New(serviceConfig)
+			if err != nil {
+				panic(err)
+			}
+			go newService.Boot()
+		}
+
+		// Create a new custom server which bundles our endpoints.
+		var newServer microserver.Server
+		{
+			serverConfig := server.DefaultConfig()
+
+			serverConfig.MicroServerConfig.Logger = newLogger
+			serverConfig.MicroServerConfig.ServiceName = name
+			serverConfig.MicroServerConfig.Viper = v
+			serverConfig.Service = newService
+
+			newServer, err = server.New(serverConfig)
+			if err != nil {
+				panic(err)
+			}
+		}
+
+		return newServer
+	}
+
+	// Create a new microkit command which manages our custom microservice.
+	var newCommand command.Command
+	{
+		commandConfig := command.DefaultConfig()
+
+		commandConfig.Logger = newLogger
+		commandConfig.ServerFactory = newServerFactory
+
+		commandConfig.Description = description
+		commandConfig.GitCommit = gitCommit
+		commandConfig.Name = name
+		commandConfig.Source = source
+
+		newCommand, err = command.New(commandConfig)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
+	daemonCommand := newCommand.DaemonCommand().CobraCommand()
+
+	daemonCommand.PersistentFlags().String(f.Service.Eventer.Environment, "", "Which environment to use for event management.")
+	daemonCommand.PersistentFlags().String(f.Service.Eventer.GitHub.OAuthToken, "", "OAuth token for authenticating against GitHub. Needs 'repo_deployment' scope.")
+	daemonCommand.PersistentFlags().String(f.Service.Eventer.GitHub.Organisation, "", "Organisation under which to check for deployments.")
+	daemonCommand.PersistentFlags().Duration(f.Service.Eventer.GitHub.PollInterval, 1*time.Minute, "Interval to poll for new deployments.")
+	daemonCommand.PersistentFlags().String(f.Service.Eventer.GitHub.ProjectList, "", "Comma seperated list of GitHub projects to check for deployments.")
+	daemonCommand.PersistentFlags().String(f.Service.Eventer.Type, string(github.GithubEventerType), "Which eventer to use for event management.")
+
+	daemonCommand.PersistentFlags().Duration(f.Service.HTTPClient.Timeout, 10*time.Second, "Timeout for HTTP requests.")
```

Comment:
The flags are actually similar to what we have in `draughtsman`. Mostly copy/paste.